python/lsst/daf/persistence/butler.py

```diff
@@ -440,13 +525,11 @@ def dataRef(self, datasetType, level=None, dataId={}, **rest):
         """
 
         datasetType = self._resolveDatasetTypeAlias(datasetType)
+        dataId = DataId(dataId)
         subset = self.subset(datasetType, level, dataId, **rest)
         if len(subset) != 1:
-            raise RuntimeError, """No unique dataset for:
-    Dataset type = %s
-    Level = %s
-    Data ID = %s
-    Keywords = %s""" % (str(datasetType), str(level), str(dataId), str(rest))
+            raise RuntimeError("No unique dataset for: Dataset type:%s Level:%s Data ID:%s Keywords:%s" % 
```

Comment:
The multi-line exception may be easier to read than the single-line form, but I'm OK with either.